lib/Zod.ts

```diff
@@ -61,7 +61,7 @@ declare module "zod" {
     function optionalParseable<Input, Output extends Constructor>(
       constructor: Output,
       parseable: (input: Input) => InstanceType<Output>
-    ): ReturnType<typeof optionalParse<Input, InstanceType<Output>>>
+    ): z.ZodType<InstanceType<Output>, z.ZodTypeDef, Input> 
```

Comment:
Needed to infer the proper return type from optionalParseable

![image](https://github.com/user-attachments/assets/ff01504b-7a57-4224-87ec-8d8e9adb0980)